webapps/frontend/app/viewmodels/wizards/addvdisk/gather.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2014 iNuron NV
+//
+// Licensed under the Open vStorage Modified Apache License (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.openvstorage.org/license
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/*global define */
+define([
+    'jquery', 'knockout', 'ovs/api', 'ovs/generic', './data', '../../containers/vpool', '../../containers/storagerouter'
+], function($, ko, api, generic, data, VPool, StorageRouter) {
+    "use strict";
+    return function() {
+        var self = this;
+
+        // Variables
+        self.data   = data;
+        self.loadStorageRoutersHandle = undefined;
+        self.loadvPoolsHandle = undefined;
+
+        // Observables
+
+        // Computed
+        self.canContinue = ko.computed(function() {
+            var valid = true, showErrors = false, reasons = [], fields = [];
+            if (self.data.name() === undefined || self.data.name() === '' || !self.data.name.valid()) {
+                valid = false;
+                fields.push('name');
+                reasons.push($.t('ovs:wizards.add_vdisk.gather.invalid_name'));
+            }
+            return { value: valid, showErrors: showErrors, reasons: reasons, fields: fields };
```

Comment:
Shouldn't we check here if e.g. ```self.data.vPool()``` isn't undefined, as this is used in the finish function as well.